go-controller/pkg/util/util_unit_test.go

```diff
@@ -0,0 +1,242 @@
+		},
+		{
+			desc:                    "ovs-vsctl command returns valid chassisID",
+			errExpected:             false,
+			onRetArgsExecUtilsIface: &onCallReturnArgs{"RunCmd", []string{"*mocks.Cmd", "string", "[]string", "string", "string", "string", "string", "string", "string"}, []interface{}{bytes.NewBuffer([]byte("4e98c281-f12b-4601-ab5a-a3d759fcb493")), bytes.NewBuffer([]byte("")), nil}},
+			onRetArgsKexecIface:     &onCallReturnArgs{"Command", []string{"string", "string", "string", "string", "string", "string", "string"}, []interface{}{mockCmd}},
+		},
+	}
+
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+
+			call := mockExecRunner.On(tc.onRetArgsExecUtilsIface.onCallMethodName)
+			for _, arg := range tc.onRetArgsExecUtilsIface.onCallMethodArgType {
+				call.Arguments = append(call.Arguments, mock.AnythingOfType(arg))
+			}
+			for _, ret := range tc.onRetArgsExecUtilsIface.retArgList {
+				call.ReturnArguments = append(call.ReturnArguments, ret)
+			}
+			call.Once()
+
+			ifaceCall := mockKexecIface.On(tc.onRetArgsKexecIface.onCallMethodName)
+			for _, arg := range tc.onRetArgsKexecIface.onCallMethodArgType {
+				ifaceCall.Arguments = append(ifaceCall.Arguments, mock.AnythingOfType(arg))
+			}
+			for _, ret := range tc.onRetArgsKexecIface.retArgList {
+				ifaceCall.ReturnArguments = append(ifaceCall.ReturnArguments, ret)
+			}
+			ifaceCall.Once()
+
+			ret, e := GetNodeChassisID()
+			if tc.errExpected {
+				assert.Error(t, e)
+			} else {
+				assert.Greater(t, len(ret), 0)
+			}
+			mockExecRunner.AssertExpectations(t)
+			mockCmd.AssertExpectations(t)
+		})
+	}
+}
+
+func TestUpdateNodeSwitchExcludeIPs(t *testing.T) {
+	mockKexecIface := new(mock_k8s_io_utils_exec.Interface)
+	mockExecRunner := new(mocks.ExecRunner)
+	mockCmd := new(mock_k8s_io_utils_exec.Cmd)
+	// below is defined in ovs.go
+	runCmdExecRunner = mockExecRunner
+	// note runner is defined in ovs.go file
+	runner = &execHelper{exec: mockKexecIface}
+
+	tests := []struct {
+		desc                    string
+		inpNodeName             string
+		inpSubnetStr            string
+		errExpected             bool
+		onRetArgsExecUtilsIface []onCallReturnArgs
+		onRetArgsKexecIface     []onCallReturnArgs
+	}{
+		{
+			desc:         "IPv4 CIDR, ovs-vsctl fails to list logical switch ports",
+			errExpected:  true,
+			inpNodeName:  "ovn-control-plane",
+			inpSubnetStr: "192.168.1.12/24",
+			onRetArgsExecUtilsIface: []onCallReturnArgs{
+				{"RunCmd", []string{"*mocks.Cmd", "string", "[]string", "string", "string", "string"}, []interface{}{bytes.NewBuffer([]byte("")), bytes.NewBuffer([]byte("")), fmt.Errorf("RunOVNNbctl error")}},
+			},
+			onRetArgsKexecIface: []onCallReturnArgs{
+				{"Command", []string{"string", "string", "string", "string"}, []interface{}{mockCmd}},
+			},
+		},
+		{
+			desc:         "IPv6 CIDR, ovs-vsctl fails to list logical switch ports",
```

Comment:
this says fails to list logical ports, but the errExpected is false?

Reply:
so maybe the description should be "IPv6 CIRD, never excludes" and check for nil?

Reply:
> so maybe the description should be "IPv6 CIDR, never excludes" and check for nil?

Looks like the intention of the code is to not to exclude any IPv6 addresses by checking for IPv6 address CIDR and skip further processing. Updating the test case name to be "IPv6 CIDR, never excludes" should be good I believe